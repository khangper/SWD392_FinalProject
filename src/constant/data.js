export const API_KEY = "API_KEY";

export const FETCH_HOME_LIVESTREAM_REQUEST = "FETCH_HOME_LIVESTREAM_REQUEST";
export const FETCH_HOME_LIVESTREAM_SUCCESS = "FETCH_HOME_LIVESTREAM_SUCCESS";
export const FETCH_HOME_LIVESTREAM_FAILURE = "FETCH_HOME_LIVESTREAM_FAILURE";

export const FETCH_HOME_FEATURED_COURSES_REQUEST =
  "FETCH_HOME_FEATURED_COURSES_REQUEST";
export const FETCH_HOME_FEATURED_COURSES_SUCCESS =
  "FETCH_HOME_FEATURED_COURSES_SUCCESS";
export const FETCH_HOME_FEATURED_COURSES_FAILURE =
  "FETCH_HOME_FEATURED_COURSES_FAILURE";
// All Live Stream
export const FETCH_STREAMS_REQUEST = "FETCH_STREAMS_REQUEST";
export const FETCH_STREAMS_SUCCESS = "FETCH_STREAMS_SUCCESS";
export const FETCH_STREAMS_FAILURE = "FETCH_STREAMS_FAILURE";

//All Instructor
export const FETCH_INSTRUCTORS_REQUEST = "FETCH_INSTRUCTORS_REQUEST";
export const FETCH_INSTRUCTORS_SUCCESS = "FETCH_INSTRUCTORS_SUCCESS";
export const FETCH_INSTRUCTORS_FAILURE = "FETCH_INSTRUCTORS_FAILURE";
export const SEARCH_INSTRUCTORS_REQUEST = "SEARCH_INSTRUCTORS_REQUEST";
export const FETCH_INSTRUCTOR_BY_ID_REQUEST = "FETCH_INSTRUCTOR_BY_ID_REQUEST";

// SaveCourse.js
export const FETCH_SAVECOURSES_REQUEST = "FETCH_SAVECOURSES_REQUEST";
export const FETCH_SAVECOURSES_SUCCESS = "FETCH_SAVECOURSES_SUCCESS";
export const FETCH_SAVECOURSES_FAILURE = "FETCH_SAVECOURSES_FAILURE";
export const DELETE_SAVECOURSE_REQUEST = "DELETE_SAVECOURSE_REQUEST";
export const DELETE_SAVECOURSE_SUCCESS = "DELETE_SAVECOURSE_SUCCESS";
export const DELETE_SAVECOURSE_FAILURE = "DELETE_SAVECOURSE_FAILURE";

// Shoppingcart.js
export const ShoppingCart_API_URL =
  "https://66908916c0a7969efd9c67ed.mockapi.io/ojt-repo/ShoppingCart";
export const ADD_TO_CART = "ADD_TO_CART";
export const REMOVE_FROM_CART = "REMOVE_FROM_CART";
export const APPLY_DISCOUNT = "APPLY_DISCOUNT";
export const ADD_TO_CART_FAILURE = " ADD_TO_CART_FAILURE";
export const ADD_TO_CART_SUCCESS = "APPLY_DISCOUNT";

//MyCourse_Course
export const FETCH_COURSES_REQUEST = "FETCH_COURSES_REQUEST";
export const FETCH_COURSES_SUCCESS = "FETCH_COURSES_SUCCESS";
export const FETCH_COURSES_FAILURE = "FETCH_COURSES_FAILURE";

export const EDIT_COURSE_REQUEST = "EDIT_COURSE_REQUEST";
export const EDIT_COURSE_SUCCESS = "EDIT_COURSE_SUCCESS";
export const EDIT_COURSE_FAILURE = "EDIT_COURSE_FAILURE";

export const DELETE_COURSE_REQUEST = "DELETE_COURSE_REQUEST";
export const DELETE_COURSE_SUCCESS = "DELETE_COURSE_SUCCESS";
export const DELETE_COURSE_FAILURE = "DELETE_COURSE_FAILURE";

//UpcommingCourse_Course
export const FETCH_UPCOMMINGCOURSES_REQUEST = "FETCH_UPCOMMINGCOURSES_REQUEST";
export const FETCH_UPCOMMINGCOURSES_SUCCESS = "FETCH_UPCOMMINGCOURSES_SUCCESS";
export const FETCH_UPCOMMINGCOURSES_FAILURE = "FETCH_UPCOMMINGCOURSES_FAILURE";

export const EDIT_UPCOMMINGCOURSE_REQUEST = "EDIT_UPCOMMINGCOURSE_REQUEST";
export const EDIT_UPCOMMINGCOURSE_SUCCESS = "EDIT_UPCOMMINGCOURSE_SUCCESS";
export const EDIT_UPCOMMINGCOURSE_FAILURE = "EDIT_UPCOMMINGCOURSE_FAILURE";

export const DELETE_UPCOMMINGCOURSE_REQUEST = "DELETE_UPCOMMINGCOURSE_REQUEST";
export const DELETE_UPCOMMINGCOURSE_SUCCESS = "DELETE_UPCOMMINGCOURSE_SUCCESS";
export const DELETE_UPCOMMINGCOURSE_FAILURE = "DELETE_UPCOMMINGCOURSE_FAILURE";
