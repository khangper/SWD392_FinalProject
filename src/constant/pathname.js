export const PATH_NAME = {
  HOME: "/home",
  EXPLORE: "/explore",
  LIVE_OUTPUT: "/live_output/:id",
  ADD_LIVE_STREAM: "/add_livestream",
  ALL_LIVE_STREAM: "/all_livestream",
  ALL_INSTRUCTOR_LIST: "/allinstructorlist/:id",
  HELP: "/help",
  REPORT_HISTORY: "/reporthistory",
  SEND_FEEDBACK: "/sendfeedback",
  OTHER_INSTRUCTOR_VIEW: "/other-instructor-view/:id",
  MY_INSTRUCTOR_PROFILE: "/my-instructor-profile",
  SAVE_COURSES: "/save-courses",
  SETTING: "/setting",
  MY_COURSE: "/courses",
  LOGIN: "/login",
  FORGOT_PASSWORD: "/forgot-password",
  SIGN_UP: "/sign-up",
  SIGN_UP_STEP: "/SignUpStep",
  THANKS: "/thanks",
  ERROR_404: "/error404",
  INVOICE: "/invoice",
  PRESS: "/press",
  SEARCH_RESULT: "/search_result",
  PAID_MEMBERSHIP: "/paidmembership",
  ABOUT_US: "/about-us",
  OUR_BLOG: "/our-blog",
  BLOG_SINGLE_VIEW: "/blog-single-view",
  CHECKOUT: "/checkout",
  SHOPPING_CART: "/shoppingcart",
  HELP_VIEW_1: "/helpview1",
  HELP_VIEW_2: "/helpview2",
  COMPANY_DETAILS: "/company-details",
  CAREER: "/career",
  APPLY_JOB: "/apply-job",
  CONTACT_US: "/contactus",
  TERMS_OF_USE: "/termsofuse",
  CERTIFICATION_FILL_FORM: "/certificationfillform",
  CERTIFICATION_TEST: "/certificates-test",
  TEST_RESULT: "/test-result",
  MESSAGE: "/messages",
  COURSES_DETAIL_VIEW: "/courses-detail-view/:id",
  PREVIEWS: "/reviews",
  COMING_SOON: "/coming-soon",
  INS_DASHBOARD: "/ins-dashboard",
  PURCHASE_COURSE: "/PurchaseCourse",
  NOTIFICATON: "/notifications",
  DOWLOADCOURSE: "/dowloadcourse",
  CERTIFICATION_CENTER: "/certificationcenter",
  SITEMAP: "/sitemap",
  EARNING: "/earning",
  PAYOUT: "/payout",
  VERTIFICATION: "/vertification",
  MYCERTIFICATIONS: "/mycertifications",
  CREDIT: "/credits",
  STATEMENTIS: "/statementis",
  STATEMENTST: "/statementst",
  ANALYTICS: "/analytics",
  CREATE_NEW_COURSE: "/create-course",
  STD_DASHBOARD: "/student-dashboard",
  NOTIFICATIONSTUDENT: "/notificationsstudent",
};
