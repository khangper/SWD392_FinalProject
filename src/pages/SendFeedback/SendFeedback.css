.feedback-container {
    padding: 20px;
}

.SendFeedback-title{
    font-size: 20px;
    margin-top: 0;
    margin-bottom: 40px;
    font-weight: 500;
    font-family: 'Roboto', sans-serif;
    color: #333;
}

.feedback-grid {
    display: grid;
    grid-template-columns: 1fr;
    grid-gap: 20px;
}

.feedback-form {
    display: grid;
    grid-template-columns: 1fr;
    grid-gap: 20px;
}

@media (min-width: 768px) {
    .feedback-grid {
        grid-template-columns: 1fr 1fr;
    }

    .feedback-form {
        grid-template-columns: 1fr;
    }
}

.ui.search.focus {
    display: flex;
    flex-direction: column;
}

.ui.form.swdh30 {
    display: flex;
    flex-direction: column;
}

.form-group1 {
    display: flex;
    flex-direction: column;
}

.image-upload-wrap {
    border: 2px dashed #d3d3d3;
    position: relative;
    padding: 20px;
    text-align: center;
    cursor: pointer;
    padding: 33px;
    width: 75%;
}

.file-upload-input {
    opacity: 0;
    position: absolute;
    top: 0;
    right: 0;
    bottom: 0;
    left: 0;
    width: 100%;
    height: 100%;
    cursor: pointer;
}

.drag-text {
    display: flex;
    flex-direction: column;
    align-items: center;
}

.SendFeedback_btn{
    height: 40px;
    padding: 0 20px;
    border: 0;
    margin-top: 45px;
    margin-bottom: 20px;
    color: #fff;
    border-radius: 3px;
    font-family: 'Roboto', sans-serif;
    font-weight: 500;
    background: #F9BF4B;
    display: block;
    width: 27%;
}

.SendFeedback_btn:hover{
    background: #e3a62b;
}

.Email-Sendfeeback{
    width: 75%;
    padding-left: 1em !important;
    padding-right: 2em !important;
    height: 40px !important;
    font-size: 14px;
    font-weight: 400;
    font-family: 'Roboto', sans-serif;
}

textarea#id_about {
    width: 78%;
    height: 139%;
}

.add-creen {
    margin-top: 40px;

    margin-bottom: 28px;
}

.form-group1 label {
    color: #333;
    font-family: 'Roboto', sans-serif;
    font-size: 14px;
    font-weight: 500;
    margin-bottom: 14px;
}


.Feedback_title{
    font-size: 20px;
    margin-top: 0;
    margin-bottom: 40px;
    font-weight: 500;
    font-family: 'Roboto', sans-serif;
    color: #333;
}

.image-upload-wrap {
    position: relative;
    border: 2px dashed #d3d3d3;
    padding: 20px;
    text-align: center;
    cursor: pointer;
    transition: border-color 0.3s;
    margin-bottom: 20px;
  }
  
  .image-upload-wrap:hover {
    border-color: #999;
  }
  
  .image-upload-wrap .file-upload-input {
    position: absolute;
    top: 0;
    left: 0;
    width: 100%;
    height: 100%;
    opacity: 0;
    cursor: pointer;
  }
  
  .image-upload-wrap .drag-text {
    margin-top: 20px;
  }
  
  .image-preview {
    width: 100%;
    height: auto;
    margin-top: 20px;
  }
  
  .image-preview img {
    max-width: 100%;
    max-height: 400px; 
    display: block;
    margin: 0 auto;
  }