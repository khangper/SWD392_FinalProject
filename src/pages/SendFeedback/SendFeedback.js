import React, { useState } from 'react';
import "./SendFeedback.css";
import send_feedback from '../../assets/send-feedback.png';
import emailjs from '@emailjs/browser';

export default function SendFeedback() {
    const [email, setEmail] = useState('');
    const [description, setDescription] = useState('');
    const [loading, setLoading] = useState(false);
    const [file, setFile] = useState(null);
    const [fileUrl, setFileUrl] = useState('');
    const [fileSelected, setFileSelected] = useState(false);
    const [imagePreviewUrl, setImagePreviewUrl] = useState(null);

    const uploadFileToCloudinary = async (file) => {
        const formData = new FormData();    
        formData.append('file', file);
        formData.append('upload_preset', 'attachment'); 

        try {
            const response = await fetch('https://api.cloudinary.com/v1_1/drpal7o6y/upload', {
                method: 'POST',
                body: formData,
            });

            const data = await response.json();
            if (response.ok) {
                return data.secure_url;
            } else {
                throw new Error(data.error.message);
            }
        } catch (error) {
            console.error('Error uploading file to Cloudinary:', error);
            return null;
        }
    };

    const handleSubmit = async (e) => {
        e.preventDefault();
        setLoading(true);

        let fileUploadedUrl = fileUrl;

        if (file) {
            fileUploadedUrl = await uploadFileToCloudinary(file);
            if (!fileUploadedUrl) {
                setLoading(false);
                return;
            }
            setFileUrl(fileUploadedUrl);
        }

        const templateParams = {
            from_email: email,
            message: description,
            attachment_url: fileUploadedUrl,
        };

        emailjs.send('service_mail_haonn', 'template_haonn', templateParams, 'ZvSJNf8vhDCcwsC_Y')
            .then((result) => {
                setLoading(false);
                setEmail('');
                setDescription('');
                setFile(null);
                setFileUrl('');
                setFileSelected(false);
                setImagePreviewUrl(null);
                console.log('Email sent successfully:', result.text);
                alert('Successfully created a new Course! Please redirect back to Explore page to review');
            }, (error) => {
                console.error('Error sending email:', error);
                setLoading(false);
            });
    };

    const handleFileChange = (e) => {
        const selectedFile = e.target.files[0];
        if (selectedFile && selectedFile.size > 10 * 1024 * 1024) { 
            alert('File size should not exceed 10MB');
        } else {
            setFile(selectedFile);
            setFileSelected(true);

            const reader = new FileReader();
            reader.onloadend = () => {
                setImagePreviewUrl(reader.result);
            };
            if (selectedFile) {
                reader.readAsDataURL(selectedFile);
            }
        }
    };

    return (
        <div className="feedback-container">
            <h2 className="Feedback_title">
                <img className="sidebar-icon" src={send_feedback} alt="Send Feedback Icon" />  Send Feedback
            </h2>
            <div className="feedback-grid">
                <form className="feedback-form" onSubmit={handleSubmit}>
                    <div className="feedback-email-from">
                        <div className="email-address">
                            <input 
                                className="Email-Sendfeeback" 
                                type="email" 
                                name="from_email" 
                                id="id_email" 
                                required 
                                maxLength="64" 
                                placeholder="Email address"
                                value={email}
                                onChange={(e) => setEmail(e.target.value)} 
                            />
                        </div>
                    </div>
                    <div className="Describe-feedback">
                        <div className="ui form swdh30">
                            <div className="field">
                                <textarea 
                                    rows="6" 
                                    name="message" 
                                    id="id_about" 
                                    placeholder="Describe your issue or share your ideas"
                                    value={description}
                                    onChange={(e) => setDescription(e.target.value)}
                                ></textarea>
                            </div>
                        </div>
                    </div>
                    <div className="form-group1 mt-30">
                        <label className="add-creen" htmlFor="file5">Add Screenshots</label>
                        <div className="image-upload-wrap">
                            <input 
                                className="file-upload-input" 
                                id="file5" 
                                name='attachment_url' 
                                type="file" 
                                accept="image/*" 
                                onChange={handleFileChange} 
                            />
                            <div className="drag-text">
                                <i className="fas fa-cloud-upload-alt"></i>
                                <h4>{fileSelected ? "Screenshot selected" : "Select screenshots to upload"}</h4>
                                <p>or drag and drop screenshots</p>
                            </div>
                            {imagePreviewUrl && (
                                <div className="image-preview">
                                    <img src={imagePreviewUrl} alt="Selected Screenshot" />
                                </div>
                            )}
                        </div>
                    </div>
                    <button 
                        className="SendFeedback_btn" 
                        type="submit" 
                        disabled={loading}
                    >
                        {loading ? 'Loading...' : 'Send Feedback'}
                    </button>
                </form>
            </div>
        </div>
    );
}