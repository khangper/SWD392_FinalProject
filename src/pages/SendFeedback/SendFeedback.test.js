import React from "react";
import { render, screen } from "@testing-library/react";
import SendFeedback from "../SendFeedback/SendFeedback";
import "@testing-library/jest-dom/extend-expect";
import { renderWithProviders } from "../../constant/test-util";

describe("Explore", () => {
  beforeAll(() => {
    jest.spyOn(console, "error").mockImplementation(() => {});
    jest.spyOn(console, "log").mockImplementation(() => {});
  });

  afterAll(() => {
    console.error.mockRestore();
    console.log.mockRestore();
  });
  it("renders the Explore component", async () => {
    const { container } = renderWithProviders(<SendFeedback />);
    expect(container).toBeDefined();
    expect(container).toMatchSnapshot();
  });
});
