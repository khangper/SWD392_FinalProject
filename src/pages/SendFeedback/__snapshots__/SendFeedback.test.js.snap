// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SendFeedback Component renders the SendFeedback component 1`] = `
<div>
  <div
    class="feedback-container"
  >
    <h2
      class="Feedback_title"
    >
      <img
        alt="Send Feedback Icon"
        class="sidebar-icon"
        src="send-feedback.png"
      />
        Send Feedback
    </h2>
    <div
      class="feedback-grid"
    >
      <form
        class="feedback-form"
      >
        <div
          class="feedback-email-from"
        >
          <div
            class="email-address"
          >
            <input
              class="Email-Sendfeeback"
              id="id_email"
              maxlength="64"
              name="from_email"
              placeholder="Email address"
              required=""
              type="email"
              value=""
            />
          </div>
        </div>
        <div
          class="Describe-feedback"
        >
          <div
            class="ui form swdh30"
          >
            <div
              class="field"
            >
              <textarea
                id="id_about"
                name="message"
                placeholder="Describe your issue or share your ideas"
                rows="6"
              />
            </div>
          </div>
        </div>
        <div
          class="form-group1 mt-30"
        >
          <label
            class="add-creen"
            for="file5"
          >
            Add Screenshots
          </label>
          <div
            class="image-upload-wrap"
          >
            <input
              accept="image/*"
              class="file-upload-input"
              id="file5"
              name="attachment_url"
              type="file"
            />
            <div
              class="drag-text"
            >
              <i
                class="fas fa-cloud-upload-alt"
              />
              <h4>
                Select screenshots to upload
              </h4>
              <p>
                or drag and drop screenshots
              </p>
            </div>
          </div>
        </div>
        <button
          class="SendFeedback_btn"
          type="submit"
        >
          Send Feedback
        </button>
      </form>
    </div>
  </div>
</div>
`;
